vmm/src/memory_manager.rs

```diff
@@ -1016,12 +1017,15 @@ impl MemoryManager {
         #[cfg(feature = "tdx")]
         let log_dirty = !tdx_enabled; // Cannot log dirty pages on a TD
 
+        let end_of_ram_area = start_of_device_area.unchecked_sub(1);
```

Comment:
It's very subtle here but it's important the start of device area hasn't been updated by `sgx_setup()`.
So I think this deserve a comment about the importance of calculating the end of RAM area soon enough.

Reply:
Weird,  I can see a comment to that affect. But in your reviewed version it wasn't there, but it is on the source view.....

Reply:
I figured out that the comment wasn't in the most appropriate commit so i've been able to modify the history to fix that.